### DISCUSSION TITLE
Add default event "connection"

### DISCUSSION THREAD

**Discussion by sikelio** [Totodore/socketioxide#294] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Currently the library doesn't include a native event like "connection" like the node js lib

**Describe the solution you'd like**
An implementation of the "connection" event for getting informations when a user connects to the websocket instance.
```js
io.on("connection", (socket) => {
  console.log(socket.id); // ojIckSD2jqNzOqIrAGzL
});
```
It could be useful like getting in realtime the users connected to the instance and not getting the socket id when a specific event is triggered.
I could be interesting to get the user IP, MAC or device name for administration on this event.

**Describe alternatives you've considered**
Parsing information from every triggered event but it's not really realtime because it needs that a user sends an event.

**Comment by Totodore** [comment, marked as answer]
In the socket.io API, the connection event is an alias of the connect event. Which corresponds to a connection of a socket to a namespace (the default in the example you gave).

https://docs.rs/socketioxide/latest/socketioxide/handler/connect/index.html

**Reply by sikelio** [reply]
Oh thanks man ^^
I didn't saw it because i'm actually learning rust :/
